### PR TITLE
Return Knative Service URLs along Istio IP

### PR DESCRIPTION
**Pull request by bbrowning**
Not all Knative installations use Istio for ingress and not all
Knative installations that use Istio expose the Istio ingress gateway.

Knative Service URLs (`.status.url`) will now be returned via the
config object in addition to Istio ingress IP so that the CLI can use
both pieces of information to do the right thing in all cases.

### CODE REVIEW

**Comment by bbrowning** [comment]
This fixes #1. For backwards compatibility and for cases where Knative installations are not properly configured with resolvable domain names for the Knative Services, this retains the Istio Ingress IP lookup code. But, it adds some error handling when doing that to ignore errors if the Istio Ingress IP could not be found because of insufficient permissions or because the Istio ingress gateways do not exist.

**Comment by billfine** [comment]
@eahefnawy can you please review and merge.

File: serverless.js [hunk 1]
```diff
@@ -74,8 +80,32 @@ class KnativeServing extends Component {
 
   // "private" methods
   async getIstioIngressIp(k8s) {
-    const res = await k8s.readNamespacedService('istio-ingressgateway', 'istio-system')
-    return res.body.status.loadBalancer.ingress[0].ip
+    try {
+      const res = await k8s.readNamespacedService('istio-ingressgateway', 'istio-system')
+      return res.body.status.loadBalancer.ingress[0].ip
+    } catch (error) {
```
Comment (rhuss) [inline serverless.js:86]: that's a good first step, but eventually we should be more clever to (a) detect istio installations without loadbalancer and (b) also check for alternative gateways (like gloo or kourier). Especially an autodetect of kourier would be interesting as it becomes a valid alternative (and maybe the default) for Knative (and OpenShift serverless)

**Review by rhuss — APPROVED** [review]
Looks good to me to not fatally break when the URL can not be extracted (which can be the case for various reason). So I'd recommend to merge the PR but still continue to work to make the gateway URL detection more robust.

**Review by pmuens — APPROVED** [review]
Thanks for working on this @bbrowning :+1:

I'll merge this since we're currently working on some updates such that this component works with the new Components implementation.

LGTM :shipit:

**Pull request merged**